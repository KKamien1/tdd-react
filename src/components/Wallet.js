import React, { Component } from "react";
import { connect } from "react-redux";

export class Wallet extends Component {
  render() {
    return (
      <div>
        <h3>
          Wallet balance: <span className="balance">{this.props.balance}</span>
        </h3>{" "}
      </div>
    );
  }
}

export default connect(state => ({
  balance: state
}))(Wallet);
