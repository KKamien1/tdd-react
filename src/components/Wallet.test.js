import React from "react";
import "../setupTest";

import { shallow } from "enzyme";

import { Wallet } from "./Wallet";
import connected from "./Wallet";

const unconnected = connected.WrappedComponent;

describe("Wallet", () => {
  const props = { balance: 20 };
  const wallet = shallow(<Wallet {...props} />);

  it("render properly", () => {
    expect(wallet).toMatchSnapshot();
  });
  it("displays the balance from prpos", () => {
    expect(wallet.find(".balance").text()).toEqual("20");
  });
});
