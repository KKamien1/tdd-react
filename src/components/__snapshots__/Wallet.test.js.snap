// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wallet render properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Wallet
    balance={20}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          Wallet balance: 
          <span
            className="balance"
          >
            20
          </span>
        </h3>,
        " ",
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Wallet balance: ",
            <span
              className="balance"
            >
              20
            </span>,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Wallet balance: ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 20,
              "className": "balance",
            },
            "ref": null,
            "rendered": 20,
            "type": "span",
          },
        ],
        "type": "h3",
      },
      " ",
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            Wallet balance: 
            <span
              className="balance"
            >
              20
            </span>
          </h3>,
          " ",
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Wallet balance: ",
              <span
                className="balance"
              >
                20
              </span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Wallet balance: ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 20,
                "className": "balance",
              },
              "ref": null,
              "rendered": 20,
              "type": "span",
            },
          ],
          "type": "h3",
        },
        " ",
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
